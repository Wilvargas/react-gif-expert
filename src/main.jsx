import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from './GifExpertApp';
import './styles.css';
import { HashRouter } from 'react-router-dom';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <HashRouter>
      <GifExpertApp/>
    </HashRouter>  
  </React.StrictMode>,
);
