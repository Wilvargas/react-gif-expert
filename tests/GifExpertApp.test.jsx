import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react'
import { GifExpertApp } from "../src/GifExpertApp"


describe('Pruebas en <GifExpertApp />', () => {

    test('debe agregar una nueva categoría cuando onAddCategory se llama', () => {

        render( <GifExpertApp /> );
    
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
    
        fireEvent.input( input, { target: { value: 'Dragon Ball' } } );
        fireEvent.submit(form)
    
        expect(screen.getByText('Dragon Ball')).toBeInTheDocument()
    })

    test('no debe agregar una categoría existente', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        // Simular la entrada de un valor que ya existe
        fireEvent.input( input, { target: { value: 'OnePunch' } });

        // Simular el envío del formulario
        fireEvent.submit( form );

        // Verificar que la categoría no se haya agregado dos veces
        expect(screen.getAllByText('OnePunch').length).toBe(1);
    });

    test('debe mantener el orden de las categorías con la nueva en la parte superior', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        // Agregar una nueva categoría
        fireEvent.input( input, { target: { value: 'Naruto' } });
        fireEvent.submit( form );

        // Obtener todas las categorías en orden
        const categories = screen.getAllByRole('heading', { level: 3 });

        // Verificar el orden
        expect(categories[0].textContent.trim()).toBe('Naruto');
        expect(categories[1].textContent.trim()).toBe('OnePunch');
    });

    test('debe coincidir con el snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });



})