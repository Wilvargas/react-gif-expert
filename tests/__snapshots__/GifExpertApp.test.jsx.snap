// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe coincidir con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    OnePunch
     
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
